### PR TITLE
Ocean API : Added RPC structure,  rocks db implementation with test cases.

### PR DESCRIPTION
**Pull request by nagarajm22**
## Summary
- creation of crate `ain-ocean` in `lib`  folder.
- Rust implementation of `whale-api` into crate `ain-ocean`.
- Implementation `rocks db` for `ain-ocean` for indexing. 
- Added test cases for rocks db.

## RPCs

- Added [jsonrpsee](https://github.com/paritytech/jsonrpsee) service structure  in `ain-ocean` crate.

## Storage

- `rock db` 
- Added rocks db test cases.

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [ ] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [ ] Includes consensus refactors
  - [ ] None

### CODE REVIEW

File: lib/ain-ocean/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "ain-ocean"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
```
Comment (prasannavl) [inline lib/ain-ocean/Cargo.toml:8]: We cant use these. These needs to use the workspace versions or it'll result in many conflicts versions.

File: lib/ain-ocean/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+tokio = { version = "1", features = ["full"] }
+serde = { version = "1.0", features = ["derive"] }
+tokio-serde = { version = "0.8.0", features = ["json"] }
+log = "0.4"
+futures = "0.3.29"
+async-trait = "0.1.74"
+anyhow = "1.0.75"
+jsonrpsee =  { version = "0.20.3", features = ["server", "macros", "http-client", "jsonrpsee-server"] }
```
Comment (prasannavl) [inline lib/ain-ocean/Cargo.toml:16]: jsonrpcsee version - let's keep the whole codebase in a single version and upgrade as a set. 

We'll need to upgrade our existing one up. But until then, use the previous.

**Comment by prasannavl** [comment]
Let's link this in a usable form first before merging. 

Make a simple dummy function, and have it used in the startup of defid init - and once the whole compilation works well, then we can merge.

**Comment by Jouzo** [comment]
Closing for #2728

**Pull request closed without merging**